flink-connectors/flink-connector-hbase/src/main/java/org/apache/flink/connector/hbase/source/hbaseendpoint/HBaseEndpoint.java

```diff
@@ -0,0 +1,290 @@
+import org.apache.hadoop.hbase.replication.ReplicationPeerDescription;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.AdminProtos.AdminService.BlockingInterface;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.zookeeper.RecoverableZooKeeper;
+import org.apache.hadoop.hbase.zookeeper.ZKUtil;
+import org.apache.hbase.thirdparty.com.google.protobuf.RpcController;
+import org.apache.hbase.thirdparty.com.google.protobuf.ServiceException;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.ZooDefs;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+/** Consumer of HBase WAL edits. */
+public class HBaseEndpoint implements ReplicationTargetInterface {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HBaseEndpoint.class);
+    private final String clusterKey;
+    /** The id under which the replication target is made known to the source cluster. */
+    private final String replicationPeerId;
+
+    private final Configuration hbaseConf;
+    private final RecoverableZooKeeper zooKeeper;
+    private final RpcServer rpcServer;
+    private final FutureCompletingBlockingQueue<HBaseSourceEvent> walEdits;
+    private final String hostName;
+    private final String tableName;
+    private boolean isRunning = false;
+
+    public HBaseEndpoint(byte[] serializedConfig, Properties properties)
+            throws IOException, KeeperException, InterruptedException {
+        this(HBaseConfigurationUtil.deserializeConfiguration(serializedConfig, null), properties);
+    }
+
+    public HBaseEndpoint(Configuration hbaseConf, Properties properties)
+            throws InterruptedException, KeeperException, IOException {
+        this(UUID.randomUUID().toString().substring(0, 5), hbaseConf, properties);
+    }
+
+    public HBaseEndpoint(String peerId, Configuration hbaseConf, Properties properties)
+            throws IOException, KeeperException, InterruptedException {
+        this.hbaseConf = hbaseConf;
+        this.clusterKey = peerId + "_clusterKey";
+        this.replicationPeerId = peerId;
+        this.hostName = HBaseSourceOptions.getHostName(properties);
+        this.tableName = HBaseSourceOptions.getTableName(properties);
+        int queueCapacity = HBaseSourceOptions.getEndpointQueueCapacity(properties);
+        this.walEdits = new FutureCompletingBlockingQueue<>(queueCapacity);
+
+        // Setup
+        zooKeeper = connectToZooKeeper();
+        rpcServer = createServer();
+        registerAtZooKeeper();
+    }
```

Comment:
❓ 00a880c0a60f5357f985d8fc77afab492cc8ddcf removes at least one constructor. The last constructor is currently not in use outside, but is open for future use, if reusing peerIds is used. This might be a case of premature abstraction and the visibility should be reduced.

Reply:
✅ changes in 4fb1b8d6b4bc7e6728c1f864d114fd0ba45b9228 reduce to one constructor